Sources/Java2Swift/JavaToSwift.swift

```diff
@@ -228,12 +228,27 @@ struct JavaToSwift: ParsableCommand {
 
       // Note that we will be translating this Java class, so it is a known class.
       translator.translatedClasses[javaClassName] = (translatedSwiftName, nil, true)
+
+      var classes: [JavaClass<JavaObject>?] = javaClass.getClasses()
+
+      // Go through all subclasses to find all of the classes to translate
+      while let internalClass = classes.popLast() {
+        if let internalClass {
+          let (javaName, swiftName) = names(from: internalClass.getName())
+          // If we have already been through this class, don't go through it again
+          guard translator.translatedClasses[javaName] == nil else { continue }
+          let currentClassName = swiftName
+          let currentSanitizedClassName = currentClassName.replacing("$", with: ".")
+          classes.append(contentsOf: internalClass.getClasses())
+          translator.translatedClasses[javaName] = (currentSanitizedClassName, nil, true)
+        }
```

Comment:
I don't think we should always recurse into nested classes. Rather, I think we should allow specification of nested classes in `Java2Swift.config`, and teach the Jar walker that produces `Java2Swift.config` to walk nested classes.

Reply:
I went back and forth on this a bit. I do like the idea of recurring down nested classes, due to the fact that the majority of uses of nested classes (at least in the JDK) are from the original class (i.e. methods that return nested classes from the main class)). I'm open to either, but like this as making a more complete class translation

Reply:
It does make the result more complete with less work on the config file, I agree. I have two concerns, but now I'm thinking that they aren't that important and can be addressed in the future.

One is that this means there is no way to disable the generation of the Swift import of a nested class: maybe we never need that, but at least for the moment I've avoided importing some Java classes because they trip up the translation in ways I wasn't able to immediately deal with. That said, if it became a problem, we could invent a way to do this.

The other is in how the output is structured. With this automatic recursion, the emitted file `Attributes.swift` now has the `Attributes.Name` type in it as well, i.e., the whole nested class definition is in one file. With the explicit recursion approach, `Attributes.Name` ends up in a separate file `Attributes+Name.swift`. To do that, the config file needs to list out the `Attributes$Name` class as its own entry so that the Java2SwiftPlugin can know all of the files that will be emitted. It's more tidy for Swift, but it's not strictly necessary... and it would be relatively straightforward in the future to support being explicit in this way in addition to the recursion that you've implemented. For example, we could say that if you have an entry in the Java2Swift.config file for a nested class, we skip that nested class in the recursion because it's been handled at the top level already.

That's all a very, very longwinded way of saying that I agree with you now and will likely implement the "explicit" mechanism I described on top of your PR. Let's merge away!

Reply:
I updated https://github.com/swiftlang/swift-java/pull/118 to build on top of this. The Jar reader now emits nested classes (which is great), so we have to deal with some nested classes being explicitly-specified in `Java2Swift.config` and others that are discovered through the recursion. Thank you for working on nested classes!